### PR TITLE
Complete Task6 & Task7

### PR DESCRIPTION
**Pull request by NDragneelL9**
1. Created notes table with reference to transaction
2. Created note model with validation and association
3. Created notes_controller
4. Updated path using url_for
5. Added ability to create notes while creating transaction
6. Added ability to choose date range for transactions listing
7. Added button to list transactions only with notes
8. Created tests for note model and note controller
9. Fixed amount input from cents to USD (both for create and update)
10. Added feature to create/update important transactions
11. Added feature to search for important transactions

### CODE REVIEW

File: spec/requests/notes_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    end
+  end
+
+  describe 'positive POST/PATCH/DELETE routes tests for transactions' do
+    it 'should create note' do
+      description = 'Desctiption example'
+      post url_for([personality, category, transaction, Note.new]),
+           params: { note: { description: description } }
+      expect(Note.last.description).to eq(description)
+      expect(response).to have_http_status(302)
```
Comment (e-popolamov) [inline spec/requests/notes_spec.rb:35]: status 302 for positive case? 
How i know 302 status it is for not auth users in devise

File: app/models/transaction_search.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  def scope_notes
+    set_category
+    @category.transactions.joins(:notes).distinct.where('transactions.created_at BETWEEN ? AND ?',
+                                                        @date_from, @date_to)
+  end
+
+  def scope_important
+    set_category
+    @category.transactions.where('created_at BETWEEN ? AND ? AND important = true',
```
Comment (e-popolamov) [inline app/models/transaction_search.rb]: this is work, but please use hash in where condition
example 
https://apidock.com/rails/ActiveRecord/QueryMethods/where

File: spec/requests/notes_spec.rb (hunk 1 above)
Comment (NDragneelL9) [reply spec/requests/notes_spec.rb:35]: Yeah, but after creating am redirecting user to url_for([@personality, @category]), so, i can follow_redirect! and expect status code 200 or just expect 302

File: app/models/transaction_search.rb (hunk 2 above)
Comment (NDragneelL9) [reply app/models/transaction_search.rb]: Done

**Comment by e-popolamov** [comment]
If we press several times to add note button we can create many notes
image
I think this is wrong behaivor

**Review by e-popolamov — CHANGES_REQUESTED** [review]
please fix this bug

**Comment by NDragneelL9** [comment]
Fixed

**Review by e-popolamov — APPROVED** [review]

**Pull request merged**